test/distributed/org/apache/cassandra/distributed/test/ColumnConstraintsTest.java

```diff
@@ -283,6 +284,21 @@ public void testLengthTableLevelConstraint() throws IOException
         }
     }
 
+    @Test
+    public void testNotNullTableLevelConstraint() throws IOException
```

Comment:
Nice. For completion, I think we can also add a `org.apache.cassandra.contraints.CreateTableWithColumnNotNullConstraintValidationTest` and a `org.apache.cassandra.contraints.AlterTableWithNotNullConstraintValidationTest` with parametrized tests for all the Cassandra types. I can create a patch to this PR if you don't have the time.